rust_snuba/src/mutations/clickhouse.rs

```diff
@@ -170,6 +170,76 @@ fn format_query(table: &str, batch: &MutationBatch) -> Vec<Vec<u8>> {
     ]
 }
 
+struct ClickhouseTestClient {
+    url: String,
+    table: String,
+    client: Client,
+}
+
+impl ClickhouseTestClient {
+    pub async fn new(table: String) -> anyhow::Result<Self> {
+        // hardcoding local Clickhouse settings
+        let hostname = "127.0.0.1";
+        let http_port = 9000;
+        let url = format!("http://{hostname}:{http_port}");
+
+        let client = reqwest::Client::new();
+        let test_table = format!("{table}_test_local\n");
+
+        let body =
+            format!("CREATE TABLE IF NOT EXISTS {test_table} AS {table}_local\n").into_bytes();
+
+        // use client to create a new table in local CH
+        client.post(url.clone()).body(body).send().await?;
+
+        Ok(Self {
+            url: url,
+            table: test_table,
+            client: client,
+        })
+    }
+
+    // actually run a query in the table
+    pub async fn run_query(&self, queries: Vec<Vec<u8>>) -> anyhow::Result<()> {
+        // run the vec of queries against Clickhouse
+        for q in queries {
+            self.client
```

Comment:
this needs to have session_id as an attribute, otherwise the temporary table will disappear